services/src/modules/plugins/metrics.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ApolloServerPlugin,
+  GraphQLRequestContextWillSendResponse,
+  GraphQLFieldResolverParams,
+} from 'apollo-server-plugin-base';
+import { FastifyInstance } from 'fastify';
+import { Histogram } from 'prom-client';
+
+let requestDurationHistogram: Histogram<string> | undefined;
+let resolverDurationHistogram: Histogram<string> | undefined;
+
+export function createMetricsPlugin(fastifyInstance: Pick<FastifyInstance, 'metrics'>): ApolloServerPlugin {
+  return {
+    serverWillStart() {
+      const promClient = fastifyInstance.metrics?.client;
+      requestDurationHistogram = new promClient.Histogram({
+        name: 'graphql_request_duration_seconds',
+        help: 'request duration in seconds',
+        labelNames: ['status'],
+      });
+
+      resolverDurationHistogram = new promClient.Histogram({
+        name: 'graphql_resolver_duration_seconds',
+        help: 'resolver duration in seconds',
+        labelNames: ['parentType', 'fieldName', 'status'],
+      });
+    },
+    requestDidStart() {
+      const endResponseTimer = requestDurationHistogram?.startTimer();
+      return {
+        willSendResponse(willSendResponseContext: GraphQLRequestContextWillSendResponse<unknown>) {
+          const {
+            response: { errors },
+          } = willSendResponseContext;
+          endResponseTimer?.({ status: errors ? 'error' : 'success' });
+        },
+        executionDidStart: () => ({
+          willResolveField(
```

Comment:
To reduce the number of labels, maybe should do this only for custom resolvers? We can check if `field.resolver === defaultFieldResolver`, and if so - don't measure it.

Reply:
I've added fix